src/macos/icons.rs

```diff
@@ -0,0 +1,79 @@
+use anyhow::{bail, Result};
+use cocoa::base::id;
+use objc::{class, msg_send, sel, sel_impl};
+use once_cell::sync::Lazy;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::hash::{Hash, Hasher};
+use std::path::Path;
+use std::path::PathBuf;
+use std::{collections::hash_map::Entry, sync::Mutex};
+use sysinfo::{PidExt, ProcessExt, ProcessRefreshKind, System, SystemExt};
+
+pub static ICON_CACHE: Lazy<Mutex<IconCache>> = Lazy::new(|| Mutex::new(IconCache::default()));
+
+#[derive(Default)]
+pub struct IconCache {
+    /// executable name -> icon hash
+    executables: HashMap<PathBuf, u64>,
+    /// icon hash -> png bytes
+    icons: HashMap<u64, Vec<u8>>,
```

Comment:
The reason why the Windows implementation has this two-layer cache is that converting the Windows icon to PNG is super slow, but checking if we already converted the same image is super fast.

What I'd like to see here for macOS:

1. benchmark TIFFRepresentation - how fast/slow is that?
2. benchmark TIFF -> PNG conversion - how fast/slow is that?
3. implement TIFF -> PNG conversion using the image crate instead of Swift & benchmark - how fast/slow is that compared to 2)?

Depending on the results we can decide how to proceed here. :)

You can check out https://github.com/mitmproxy/mitmproxy_rs/blob/main/benches/process.rs for Rust benchmarking examples :)